couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
```

Comment:
Is it suitable to make a property setter for `server.state['view_lib']` ?

like this
```diff
diff --git a/couchdb/server/__init__.py b/couchdb/server/__init__.py
index 40a4328..dc0e6ee 100644
--- a/couchdb/server/__init__.py
+++ b/couchdb/server/__init__.py
@@ -820,3 +820,7 @@ class SimpleQueryServer(BaseQueryServer):
     def view_lib(self):
         """Returns stored view lib which could be added by :meth:`add_lib`"""
         return self.state['view_lib']
+
+    @view_lib.setter
+    def view_lib(self, val):
+        self.state['view_lib'] = val
diff --git a/couchdb/server/state.py b/couchdb/server/state.py
index ee0ee0e..d08768d 100644
--- a/couchdb/server/state.py
+++ b/couchdb/server/state.py
@@ -29,7 +29,7 @@ def reset(server, config=None):
         log.debug('Set new query config:\n%s', config)
         server.state['query_config'].update(config)
     if server.version >= (1, 1, 0):
-        server.state['view_lib'] = ''
+        server.view_lib = ''
     return True


@@ -75,5 +75,5 @@ def add_lib(server, lib):
     .. versionadded:: 1.1.0
     """
     log.debug('Set view_lib:\n%s', lib)
-    server.state['view_lib'] = lib
+    server.view_lib = lib
     return True
diff --git a/couchdb/tests/server/state.py b/couchdb/tests/server/state.py
index 42d7f52..102bda0 100644
--- a/couchdb/tests/server/state.py
+++ b/couchdb/tests/server/state.py
@@ -52,10 +52,12 @@ class StateTestCase(unittest.TestCase):
         server.state['functions'].append('foo')
         server.state['functions_src'].append('bar')
         server.state['query_config']['baz'] = 42
+        server.state['view_lib'] = {'lib': {'magic': '42'}}
         state.reset(server)
         self.assertEqual(server.state['functions'], [])
         self.assertEqual(server.state['functions_src'], [])
         self.assertEqual(server.state['query_config'], {})
+        self.assertFalse(server.state['view_lib'])

     def test_reset_and_update_query_config(self):
         """should reset query config and set new values to it"""
```